source/_posts/2018-01-16-Testing-Vuejs-in-Rails-with-Webpacker-and-Jest.md

```diff
@@ -0,0 +1,153 @@
+{% codeblock lang:json %}
+"jest": {
+  "roots": [
+    "spec/javascript"
+  ]
+},
+{% endcodeblock %}
+
+<div style="background: linen; padding: 15px; margin-bottom: 15px; border-radius: 15px">
+  <p>
+    If your <code>package.json</code> is quite big and you don't want to keep adding more stuff to it, you can define the jest configuration through the <code>--config &lt;path/to/js|json&gt;</code> option. If you choose to do so, your <code>package.json</code> should now be like:
+  </p>
+  {% codeblock lang:json %}
+  {
+    "scripts": {
+      “test”: “jest --config spec/javascript/jest.conf.js”,
+      ...
+    },
+    ...
+  }
+  {% endcodeblock %}
+</div>
+
+To verify it worked, you can create a `spec/javascript/team.spec.js` file with a simple test like:
+{% codeblock lang:javascript %}
+test('there is no I in team', () => {
+  expect('team').not.toMatch(/I/);
+});
+{% endcodeblock %}
+Now run `yarn test` again and you should see a green "PASS" in the output meaning that it worked 🎉.
+
+## 3. Babel to the rescue
+Now that we got our first test working, let's take it one step further and try to test a Vue component.
+The first thing you'd probably try is to create a file under the `spec/javascript/` directory and name it something like `my_component.spec.js`. Then try to import your component from within your spec with the [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) statement like:
+
+{% codeblock lang:javascript %}
+  import MyComponent from '../../app/javascript/my_component.vue';
+{% endcodeblock %}
+
+If you did try this, go ahead and run your tests. You'll see a `SyntaxError: Unexpected token import` in the output.
+
+The problem here is that `import` is part of ECMAScript 6, so we need the help of a transpiler such as [Babel](https://babeljs.io/).
+To get it working you need to install two packages by running `yarn add --dev babel-jest babel-preset-es2015` and add the "es2015" preset to your `.babelrc` file:
+{% codeblock lang:json %}
+{
+  "presets": ["es2015",
+    ["env", {
+			...
+{% endcodeblock %}
+If you want to take it one step further, you can add [moduleDirectories](https://facebook.github.io/jest/docs/en/configuration.html#moduledirectories-array-string) to your `.package.json` so you don't have to type out the full path to your modules:
+{% codeblock lang:json %}
+"jest": {
+	...
+  "moduleDirectories": [
+    "node_modules",
+    "app/javascript"
+	]
+}
+{% endcodeblock %}
+So what we had before as
+{% codeblock lang:javascript %}
+  import MyComponent from '../../app/javascript/my_component.vue';
+{% endcodeblock %}
+can now be written as
+{% codeblock lang:javascript %}
+  import MyComponent from 'my_component.vue';
+{% endcodeblock %}
+
+## 4. Where's Vue?
+If you followed every step, you should be getting a `SyntaxError` when trying to run your tests. This means that it successfully imported your component, but it can't yet understand Vue syntax.
```

Comment:
I think that .vue files has a special _format_ rather than a _syntax_ (the term _syntax_ is more suitable for language rules).

I would rewrite `but it can’t yet understand Vue syntax` with `but it can’t yet understand .vue files format`